File: src/main/java/org/goafabric/core/data/logic/mapper/OrganizationMapper.java
package org.goafabric.core.data.logic.mapper;

import org.goafabric.core.data.controller.vo.Organization;
import org.goafabric.core.data.repository.entity.OrganizationEo;
import org.mapstruct.Mapper;
import org.mapstruct.ReportingPolicy;

import java.util.List;


@Mapper(componentModel = "jakarta", unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface OrganizationMapper extends WorkaroundMapper {
    Organization map(OrganizationEo value);

    OrganizationEo map(Organization value);

    List<Organization> map(List<OrganizationEo> value);
}

File: src/main/kotlin/org/goafabric/core/organization/logic/mapper/OrganizationMapper.kt
package org.goafabric.core.organization.logic.mapper

import org.goafabric.core.organization.controller.dto.Organization
import org.goafabric.core.organization.persistence.entity.OrganizationEo
import org.mapstruct.Mapper
import org.mapstruct.ReportingPolicy

@Mapper(componentModel = "cdi", unmappedTargetPolicy = ReportingPolicy.IGNORE, uses = [AddressMapper::class])
interface OrganizationMapper {
    fun map(value: OrganizationEo): Organization
    fun map(value: Organization): OrganizationEo
    fun map(values: List<OrganizationEo>): List<Organization>
}
